backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java

```diff
@@ -157,6 +157,8 @@ public DatasetVersion createDatasetVersion(
         DatasetVersionEntity datasetVersionEntity =
             RdbmsUtils.generateDatasetVersionEntity(datasetVersion);
         session.save(datasetVersionEntity);
+        setDatasetUpdateTime(session, Collections.singletonList(datasetVersion.getDatasetId()));
+        transaction.commit();
```

Comment:
Always create the role binding before creating the object, and always delete the object before deleting the bindings. We can have the system giving permission for an object that doesn't exist, but we can't have the system having an object with no permission.

Reply:
@conradoverta  `Always create the role binding before creating the object,` is not possible for all entities because for the project we are creating ID our self it will be fine to create before creating it but like repository entity, we can not do it before creating because repository id is auto genearted by the Hibernate so we got id after entity creation. For role binding, we need to add entity id so in that case, we should create role binding after entity creation.

Reply:
That will cause issues. If we don't satisfy the rule I wrote, the code will have a bug waiting to happen. We'll have to figure out how to resolve it.

Reply:
we can create the id here and pass it to be used at https://github.com/VertaAI/modeldb/blob/master/backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAO.java#L330

Reply:
Modified changes